rewrite-java/src/main/java/org/openrewrite/java/JavaTemplate.java

```diff
@@ -27,25 +27,53 @@
 import org.openrewrite.java.tree.JavaCoordinates;
 import org.openrewrite.template.SourceTemplate;
 
+import java.io.IOException;
+import java.io.InputStream;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
-import java.util.HashSet;
-import java.util.Set;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
 import java.util.function.Consumer;
 
 @SuppressWarnings("unused")
 public class JavaTemplate implements SourceTemplate<J, JavaCoordinates> {
+    protected static final Path TEMPLATE_CLASSPATH_DIR;
+
+    static {
+        try {
+            TEMPLATE_CLASSPATH_DIR = Files.createTempDirectory("java-template");
+            Path templateDir = Files.createDirectories(TEMPLATE_CLASSPATH_DIR.resolve("org/openrewrite/java/internal/template"));
```

Comment:
File IO in a static initializer feels unfortunate. But I do like using precomplied `__M__` and `__P__` to get around language differences. Smart

Reply:
I could do a lazy initialization as well at runtime.